BootloaderCommonPkg/Library/SecureBootLib/SecureBootHash.c

```diff
@@ -30,22 +33,31 @@ RETURN_STATUS
 DoHashVerify (
   IN CONST UINT8           *Data,
   IN       UINT32           Length,
-  IN       UINT8            ComponentType
+  IN       UINT8            ComponentType,
+  IN OUT   UINT8           *Hash
```

Comment:
We can add new API for this.  But I prefer changing the existing one to avoid too many APIs left behind. 

For DoHashVerify,  I think we can add a HashAlgorithm parameter to support different hash type.

Reply:
agree changing existing API is a clear way. But at some point, we need make sure SBL could be backward compatible in APIs to make sure customer could upgrade SblOpen smoothly. 
For the API we plan to drop, we could print a warning/usage to let user to use new API. then after some time (e.g. a year) we could bump SBL version and drop old API.

Reply:
In this case,  It should not be a big issue.   The API is in SecureBootLib.  And all secure boot related flow are controlled by the core code.  As long as the core code is updated properly,  it should just work fine.